examples/kg-chat/dependency_chatbot.py

```diff
@@ -0,0 +1,222 @@
+from langroid.utils.constants import NO_ANSWER
+from langroid.agent.chat_agent import ChatAgent, ChatAgentConfig
+from langroid.utils.configuration import set_global, Settings
+from langroid.utils.logging import setup_colored_logging
+from langroid.agent.tool_message import ToolMessage
+from langroid.agent.tools.google_search_tool import GoogleSearchTool
+
+from langroid.agent.task import Task
+from cypher_message import CONSTRUCT_DEPENDENCY_GRAPH
+
+
+app = typer.Typer()
+
+setup_colored_logging()
+
+
+class GetPackageInfo(ToolMessage):
+    request = "construct_dependency_graph"
+    purpose = f"""Get package <package_version>, <package_type>, and <package_name>.
+    For the <package_version>, obtain the recent version and should be a number. 
+    For the <package_type>, return if the package is PyPI or not.
+      Otherwise, return {NO_ANSWER}.
+    For the <package_name>, return the package name provided by the user.
+    ALL strings are in small letter. 
+    """
+    package_version: str
+    package_type: str
+    package_name: str
+
+
+class DependencyGraphAgent(Neo4jChatAgent):
+    package_name: str
+
+    def construct_dependency_graph(self, msg: GetPackageInfo) -> None:
+        check_db_exist = (
+            "MATCH (n) WHERE n.name = $name AND n.version = $version RETURN n LIMIT 1"
+        )
+        response = self.read_query(
+            check_db_exist, {"name": msg.package_name, "version": msg.package_version}
+        )
+        if "No records found" not in response:
+            self.config.database_created = True
+            return "Database Exists"
+        else:
+            construct_dependency_graph = CONSTRUCT_DEPENDENCY_GRAPH.format(
+                package_type=msg.package_type,
+                package_name=msg.package_name,
+                package_version=msg.package_version,
+            )
+            if self.write_query(construct_dependency_graph):
+                self.config.database_created = True
+                return "Database is created!"
+            else:
+                return f"""
+                    Database is not created!
+                    Seems the package {msg.package_name} is not found,
+                    """
+
+
+class CLIOptions(BaseSettings):
+    model: str = ""
+    fn_api: bool = False
+
+    class Config:
+        extra = "forbid"
+        env_prefix = ""
+
+
+def chat(opts: CLIOptions) -> None:
+    print(
+        """
+        [blue]Welcome to Dependency Analysis chatbot!
+        Enter x or q to quit at any point.
+        """
+    )
+
+    load_dotenv()
+
+    dependency_agent = DependencyGraphAgent(
+        config=Neo4jChatAgentConfig(
+            uri=os.getenv("NEO4J_URI"),
+            username=os.getenv("NEO4J_USERNAME"),
+            password=os.getenv("NEO4J_PASSWORD"),
+            database=os.getenv("NEO4J_DATABASE"),
+            use_tools=opts.fn_api,
+            use_functions_api=not opts.fn_api,
```

Comment:
let's have it use fn api by default. When I ran it with no options it was using langroid tools.